axlearn/common/compiler_options.py

```diff
@@ -52,6 +52,11 @@ def default_xla_options(
             # concurrently with gradient computation for the following layer.
             xla_tpu_enable_data_parallel_all_reduce_opt="true",
             xla_tpu_data_parallel_opt_different_sized_ops="true",
+            # If MegaScale Runtime Error is encountered when running multi-slice jobs,
+            # enabling this flag will allow for termination of the job, triggering
+            # the process to exit. This is set to true to prevent the job from
+            # silently hanging and to reduce time to recovery.
+            megascale_abort_on_hangs="true",
         )
 
     # Validate options. Will never fail if this function is implemented correctly.
```

Comment:
Should we also set:
```
        --megascale_crash_on_stale_unmatched_keys=True
        --megascale_callback_registry_stale_threshold=10m
```
(Directly in `launch.py`, not here, for the aot compatibility reasons described in the other comment.)

Reply:
CC @tgunter for any thoughts about megascale flags.

Reply:
> Should we also set:
> 
> ```
>         --megascale_crash_on_stale_unmatched_keys=True
>         --megascale_callback_registry_stale_threshold=10m
> ```
> 
> (Directly in `launch.py`, not here, for the aot compatibility reasons described in the other comment.)

These 2 flags has been retried in 2024/4/24. They have no effect for libtpu built after that date.

Reply:
Thanks for the explanation!